lib/derivations.nix

```diff
@@ -206,4 +206,40 @@ in
   optionalDrvAttr =
     cond:
     value: if cond then value else null;
+
+  /**
+    Wrap a derivation such that instantiating it produces a warning.
+
```

Comment:
Maybe also wrap `override` and `overrideAttrs` if they exist? That doesn't cover all possible overriding functions (`passthru.withFoo = finalAttrs.finalPackage.overrideAttrs ...;`) but that's quite rare.

Perhaps it'd be better to cover all attributes except like `meta`, `name` and `type`?

Regardless of what works best, it would be good to document how it works.

Reply:
Changed it to wrap everything except those three attributes. Also added a sentence on what exactly it does, and a hint that it's related to `nix search`.